cmake/git-versioning.cmake

```diff
@@ -0,0 +1,124 @@
+macro(git_command args output_variable)
+    execute_process(
+            COMMAND ${GIT_EXECUTABLE} ${args}
+            WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+            RESULT_VARIABLE exit_status
+            ERROR_VARIABLE error_output
+            OUTPUT_VARIABLE temp_variable
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+    if (NOT ${exit_status} MATCHES "0")
+        message(SEND_ERROR "Unable to retrieve ${output_variable}.")
+        message(FATAL_ERROR ${error_output})
+    endif ()
+    set(${output_variable} "${temp_variable}")
+endmacro()
+
+macro(process_git_version)
+
+    find_package(Git)
```

Comment:
Is the find_package(Git) still needed on the top-level CMakeLists.txt? If not, that could be removed there.

Reply:
We need git for submodule initialization too. These scripts are designed so that they can work stand-alone for any project (not just our atheneum projects). I don't think there is any harm to including it twice, though maybe both should be in their respective CMake scripts where git is invoked instead of at the top level. I'll look into it.